File: Java/Fibonacci.java
class fibonacci {
    public static long fib(int x) {
        if(x<2)
        return x;
        int a=0,b=1,c=0;
        for(int i=2;i<=x;i++){
            c=a+b;
            a=b;
            b=c;
        }
        return c;
    }

    public static void main(String args[]) {
        int n=9;
        System.out.println("The "+n+"th Fibonacci number is: "+fib(n));
        
    }
}

File: Haskell/Fibonacci.hs


fibonacci :: Integer -> Integer
fibonacci 0 = 0
fibonacci 1 = 1
fibonacci n = fibonacci (n-1) + fibonacci (n-2)


Prelude> :load Fib.hs
[1 of 1] Compiling Main             ( Fib.hs, interpreted )
Ok, modules loaded: Main.
*Main> 
*Main> fibonacci 0
0
*Main> fibonacci 1
1
*Main> fibonacci 2
1
*Main> fibonacci 3
2
*Main> fibonacci 4
3
*Main> fibonacci 5
5

